src/core/proj/qgscoordinatereferencesystem.cpp

```diff
@@ -2961,6 +2961,39 @@ QgsCoordinateReferenceSystem QgsCoordinateReferenceSystem::toGeographicCrs() con
   }
 }
 
+QgsCoordinateReferenceSystem QgsCoordinateReferenceSystem::verticalCrs() const
+{
+  switch ( type() )
+  {
+    case Qgis::CrsType::Unknown:
+    case Qgis::CrsType::Geodetic:
+    case Qgis::CrsType::Geocentric:
+    case Qgis::CrsType::Geographic2d:
+    case Qgis::CrsType::Projected:
+    case Qgis::CrsType::Temporal:
+    case Qgis::CrsType::Engineering:
+    case Qgis::CrsType::Bound:
+    case Qgis::CrsType::Other:
+    case Qgis::CrsType::DerivedProjected:
+      return QgsCoordinateReferenceSystem();
+
+    case Qgis::CrsType::Geographic3d:
+    case Qgis::CrsType::Vertical:
+      return *this;
```

Comment:
A (ISO19111 certifiied) geodesist will frown upon returing a Geographic3D CRS as a vertical CRS... One expectation is that a vertical CRS has one single axis. There might be reason to return a Geographic3D CRS in QGIS, but this should be clearly documented. What are the reasons for doing that?

Reply:
@rouault 

My thinking was that using a geographic 3d system implies the vertical reference itself. Is there a scenario where you'd have a geographic 3d system + an explicit vertical crs?

Reply:
no, a geographic 3d CRS cannot be accompanied with a vertical CRS. My point was mostly that a verticalCrs() method returning a Geographic 3D CRS is not really standard compliant (PROJ wouldn't do that for example). The official definition of a vertical CRS implies it has a single axis, and that it is gravity related...
To workaround that a bit and providing "vertical CRS" with ellipsoidal height, ESRI has come with things like
```
VERTCRS["WGS_1984",
    VDATUM["D_WGS_1984"],
    CS[vertical,1],
        AXIS["ellipsoidal height (h)",up,
            LENGTHUNIT["metre",1]],
    USAGE[
        SCOPE["Not known."],
        AREA["World."],
        BBOX[-90,-180,90,180]],
    ID["ESRI",115700]]
```
but I'm not sure we'd want to promote that too

Reply:
Ok, I'll make this return an invalid crs for geographic 3d here then, and handle the geographic 3d case outside of this method 👍

Reply:
(done)